### PR TITLE
8293587: Fix mistakes in FX API docs

### PR DESCRIPTION
**Pull request by nlisker**
Fixes and cleanup in the areas in the linked issue.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8293587](https://bugs.openjdk.org/browse/JDK-8293587): Fix mistakes in FX API docs

### Reviewers
 * [John Hendrikx](https://openjdk.org/census#jhendrikx) (@hjohn - Committer)
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Andy Goryachev](https://openjdk.org/census#angorya) (@andy-goryachev-oracle - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx pull/1025/head:pull/1025` \
`$ git checkout pull/1025`

Update a local copy of the PR: \
`$ git checkout pull/1025` \
`$ git pull https://git.openjdk.org/jfx pull/1025/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1025`

View PR using the GUI difftool: \
`$ git pr show -t 1025`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1025.diff">https://git.openjdk.org/jfx/pull/1025.diff</a>

</details>

### CODE REVIEW

**Comment by nlisker** [comment]
Integrating can wait until a bit before the release to allow for more mistakes to be included.

File: modules/javafx.controls/src/main/java/javafx/scene/control/ListView.java [hunk 1]
```diff
@@ -462,17 +462,14 @@ public final ObjectProperty<ObservableList<T>> itemsProperty() {
...
-    // --- Placeholder Node
-    private ObjectProperty<Node> placeholder;
     /**
-     * This Node is shown to the user when the listview has no content to show.
-     * This may be the case because the table model has no data in the first
-     * place or that a filter has been applied to the list model, resulting
-     * in there being nothing to show the user..
-     * @return the placeholder property for this ListView
+     * The {@code Node} to show to the user when the {@code ListView} has no content to show.
+     * This happens when the table model has no data or when a filter has been applied to the list model, resulting in
```
Comment (hjohn) [inline modules/javafx.controls/src/main/java/javafx/scene/control/ListView.java:467]:
```suggestion
     * This happens when the list model has no data or when a filter has been applied to the list model, resulting in
```

File: modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java [hunk 2]
```diff
@@ -59,38 +59,38 @@
 import javafx.css.StyleableProperty;
 
 /**
- * TextFlow is special layout designed to lay out rich text.
- * It can be used to layout several {@link Text} nodes in a single text flow.
- * The TextFlow uses the text and the font of each {@link Text} node inside of it
+ * A special layout designed to lay out rich text.
```
Comment (hjohn) [inline modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java:62]: What's "special" about it? :-)
```suggestion
 * A layout designed to lay out rich text.
```

File: modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java [hunk 3]
```diff
@@ -59,38 +59,38 @@
...
+ * For example, the {@code x} and {@code y} properties of the {@code Text} node are ignored since
  * the location of the node is determined by the parent. Likewise, the wrapping
- * width in the {@link Text} node is ignored since the width used for wrapping
- * is the TextFlow's width. The value of the <code>pickOnBounds</code> property
- * of a {@link Text} is set to <code>false</code> when it is laid out by the
- * TextFlow. This happens because the content of a single {@link Text} node can
- * divided and placed in the different locations on the TextFlow (usually due to
+ * width in the {@code Text} node is ignored since the width used for wrapping
+ * is the {@code TextFlow}'s width. The value of the {@code pickOnBounds} property
+ * of a {@code Text} is set to {@code false} when it is laid out by the
```
Comment (hjohn) [inline modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java:81]:
```suggestion
 * of a {@code Text} node is set to {@code false} when it is laid out by the
```

File: modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java [hunk 4]
```diff
@@ -59,38 +59,38 @@
...
+ * {@code TextFlow}. This happens because the content of a single {@code Text} node can be
+ * divided and placed in different locations in the {@code TextFlow} (usually due to
```
Comment (hjohn) [inline modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java:83]:
```suggestion
 * split and placed in different locations in the {@code TextFlow} (usually due to
```

File: modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java [hunk 5]
```diff
@@ -59,38 +59,38 @@
...
  * line breaking and bidi reordering).
  *
  * <p>
  * The wrapping width of the layout is determined by the region's current width.
- * It can be specified by the application by setting the textflow's preferred
+ * It can be specified by the application by setting the {@code TextFlow}'s preferred
  * width. If no wrapping is desired, the application can either set the preferred
- * with to Double.MAX_VALUE or Region.USE_COMPUTED_SIZE.
+ * with to {@code Double.MAX_VALUE} or {@code Region.USE_COMPUTED_SIZE}.
```
Comment (hjohn) [inline modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java:90]: Should these be links?  They're the first mention in this doc.

File: modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java [hunk 6]
```diff
@@ -133,21 +133,21 @@
  * <td>Double.MAX_VALUE</td><td>Double.MAX_VALUE</td></tr>
```
Comment (hjohn) [inline modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java:133]:
```suggestion
 * <td>{@code Double.MAX_VALUE}</td><td>{@code Double.MAX_VALUE}</td></tr>
```

**Review by hjohn — APPROVED** [review]
Looks good to me, I added some more minor suggestions as well.

File: modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java [hunk 7]
```diff
@@ -133,21 +133,21 @@
...
+ * A {@code TextFlow}'s unbounded maximum width and height are an indication to the parent that
  * it may be resized beyond its preferred size to fill whatever space is assigned to it.
  * <p>
- * TextFlow provides properties for setting the size range directly.  These
- * properties default to the sentinel value Region.USE_COMPUTED_SIZE, however the
+ * {@code TextFlow} provides properties for setting the size range directly. These
+ * properties default to the sentinel value {@code Region.USE_COMPUTED_SIZE}, however the
  * application may set them to other values as needed:
  * <pre><code>
  *     <b>textflow.setMaxWidth(500);</b>
```
Comment (kevinrushforth) [inline modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java:143]: Maybe remove the `<b>` tag while you are at it?

**Review by kevinrushforth — COMMENTED** [review]
Other than John's comments, this looks good to me. I noted one more possible formatting change and an unrelated follow-up cleanup issue I plan to file.

File: modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java (hunk 2 above)
Comment (nlisker) [reply modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java:62]: Maybe it's special in the sense that it's specialized towards displaying text, unlike other `Pane`s that are multi-purpose for all nodes. "designed to" already points to that. Maybe "A specialized layout for rich text."?

File: modules/javafx.graphics/src/main/java/javafx/scene/Scene.java [hunk 8]
```diff
@@ -3274,7 +3274,7 @@ private boolean processRecognized(DragEvent de) {
 
         private void processDropEnd(DragEvent de) {
             if (source == null) {
-                System.out.println("Scene.DnDGesture.processDropEnd() - UNEXPECTD - source is NULL");
+                System.out.println("Scene.DnDGesture.processDropEnd() - UNEXPECTED - source is NULL");
```
[the remark this answers is not in the document]
Comment (nlisker) [reply modules/javafx.graphics/src/main/java/javafx/scene/Scene.java:3277]: I never understood when something should be logged vs. printed. And do we ever print to `out`?

File: modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java (hunk 7 above)
Comment (nlisker) [reply modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java:143]: All the other panes use this format, for example, https://openjfx.io/javadoc/19/javafx.graphics/javafx/scene/layout/BorderPane.html.

Maybe in a Panes doc cleanup issue. I always thought more images were needed in these to show the layouts, so I could look into that.

File: modules/javafx.graphics/src/main/java/javafx/scene/Scene.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (andy-goryachev-oracle) [reply modules/javafx.graphics/src/main/java/javafx/scene/Scene.java:3277]: I wish we were using logging...

File: modules/javafx.controls/src/main/java/javafx/scene/control/ListView.java [hunk 9]
```diff
@@ -462,17 +462,14 @@ public final ObjectProperty<ObservableList<T>> itemsProperty() {
...
-     * place or that a filter has been applied to the list model, resulting
-     * in there being nothing to show the user..
-     * @return the placeholder property for this ListView
+     * The {@code Node} to show to the user when the {@code ListView} has no content to show.
+     * This happens when the list model has no data or when a filter has been applied to the list model, resulting in
+     * there being nothing to show the user.
+     *
      * @since JavaFX 8.0
      */
+    private ObjectProperty<Node> placeholder;
```
Comment (andy-goryachev-oracle) [inline modules/javafx.controls/src/main/java/javafx/scene/control/ListView.java:472]: could we add a blank line after this statement?

File: modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java (hunk 2 above)
Comment (andy-goryachev-oracle) [reply modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java:62]: A specialized layout... +1

File: modules/javafx.graphics/src/main/java/javafx/scene/Scene.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (hjohn) [reply modules/javafx.graphics/src/main/java/javafx/scene/Scene.java:3277]: Some investigation is needed if source can ever be null in such cases.  I think some of these are to inform the programmer (not the user) something unexpected happens, in which case it should be an assert or exception.  If it then ever does happen, it should be fixed.

[the remark this answers is not in the document]
Comment (hjohn) [reply modules/javafx.graphics/src/main/java/javafx/scene/Scene.java:3277]: The drag/drop code is full of what looks like asserts that are just dumped to `System.err`.  It looks like left overs that helped during development of the underlying state machine, but it shouldn't get in such states normally.  When I run the tests, none of the exceptional cases or `System.err` code has any coverage which can be a sign that these cases can't be reached.

File: modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java (hunk 2 above)
Comment (nlisker) [reply modules/javafx.graphics/src/main/java/javafx/scene/text/TextFlow.java:62]:
```suggestion
 * A specialized layout for rich text.
```

**Review by andy-goryachev-oracle — APPROVED** [review]
definitely an improvement, thank you.

**Review by hjohn — APPROVED** [review]

**Review by kevinrushforth — APPROVED** [review]

**Pull request closed without merging**